### DISCUSSION TITLE
Default state reactive

### DISCUSSION THREAD

**Discussion by vincent-tarrit** [filamentphp/filament#5942] [Help]
Hello, 

I would like to set a default value (with closure) in a field and call afterStateUpdated() to render the others fields as if the default was input by human. Could you help me how to do it ? Thank you !

**Comment by danharrin** [comment]
You cant set a default value after the form is loaded. Use `$set()` to set the value of a field from a closure in `afterStateUpdated()`. Search for "closure customization" in the docs.

**Comment by vincent-tarrit** [comment]
Thank you for your answer, my problem is that it's missing a trigger to call the afterStateUpdated()... Do you think it would be technically possible to do something as 

```php
->default(function () {
    return "FOO";
}, shouldTriggerStateUpdated: true)
```

If it's possible i would be happy to try a PR. 


To put you in the context here is an example of what I would like to do :
```php
Select::make('beach_id')
    ->options(Beach::fromMyProjects()->pluck('name', 'id'))
    ->label('Beach')
    ->reactive()
    ->default(function () {
        return session('beach_id');
    })
    ->afterStateUpdated(function ($set) {
      $set('beach_name', Beach::find($set('beach_id'))->name);
  })
  ->required(),

TextInput::make('beach_name')
  ->label('Beach name')
  ->required()
  ->disabled(),
```

**Reply by danharrin** [reply]
`afterStateUpdated()` is only called when the user makes changes to the field. Add a `default()` to the beach_name field and use the beach_id from the session to fill the name.

**Reply by vincent-tarrit** [reply]
okok, so it's not what i wanted, The goal was to touch only one field. It's ok I will figure it out ! Thank you very much for your answers!
